elk/training/train.py

```diff
@@ -78,88 +78,129 @@ def train_reporter(
-
         device = self.get_device(devices, world_size)
 
-        x0, x1, val_x0, val_x1, train_gt, val_gt, val_lm_preds = self.prepare_data(
-            device, layer
-        )
-        pseudo_auroc = self.get_pseudo_auroc(layer, x0, x1, val_x0, val_x1)
+        train_dict = self.prepare_data(device, layer, "train")
+        val_dict = self.prepare_data(device, layer, "val")
+
+        # Can't figure out a way to make this line less ugly
+        hidden_size = next(iter(train_dict.values()))[0].shape[-1]
+        reporter_dir, lr_dir = self.create_models_dir(assert_type(Path, self.out_dir))
+        pseudo_clf = self.get_pseudo_classifier(train_dict, device)
 
         if isinstance(self.cfg.net, CcsReporterConfig):
-            reporter = CcsReporter(x0.shape[-1], self.cfg.net, device=device)
+            assert len(train_dict) == 1, "CCS only supports single-task training"
+
+            reporter = CcsReporter(hidden_size, self.cfg.net, device=device)
+            (x0, x1, labels, _) = next(iter(train_dict.values()))
+            train_loss = reporter.fit(x0, x1, labels)
+
         elif isinstance(self.cfg.net, EigenReporterConfig):
-            reporter = EigenReporter(x0.shape[-1], self.cfg.net, device=device)
+            # To enable training on multiple tasks with different numbers of variants,
+            # we update the statistics in a streaming fashion and then fit
+            reporter = EigenReporter(hidden_size, self.cfg.net, device=device)
+            for ds_name, (x0, x1, labels, _) in train_dict.items():
+                reporter.update(x0, x1)
+
+            train_loss = reporter.fit_streaming()
         else:
             raise ValueError(f"Unknown reporter config type: {type(self.cfg.net)}")
 
-        train_loss = reporter.fit(x0, x1, train_gt)
-        val_result = reporter.score(
-            val_gt,
-            val_x0,
-            val_x1,
-        )
+        # Save reporter checkpoint to disk
+        with open(reporter_dir / f"layer_{layer}.pt", "wb") as file:
+            torch.save(reporter, file)
 
-        reporter_dir, lr_dir = self.create_models_dir(assert_type(Path, self.out_dir))
-        if val_lm_preds is not None:
-            val_gt_cpu = val_gt.repeat_interleave(val_lm_preds.shape[1]).float().cpu()
-            val_lm_auroc = float(roc_auc_score(val_gt_cpu, val_lm_preds.flatten()))
-            val_lm_acc = float(accuracy_score(val_gt_cpu, val_lm_preds.flatten() > 0.5))
-        else:
-            val_lm_auroc = None
-            val_lm_acc = None
-
-        row = pd.Series(
-            {
-                "layer": layer,
-                "pseudo_auroc": pseudo_auroc,
-                "train_loss": train_loss,
-                **val_result._asdict(),
-                "lm_auroc": val_lm_auroc,
-                "lm_acc": val_lm_acc,
-            }
-        )
+        # Fit supervised logistic regression model
+        lr_model = train_supervised(train_dict, device=device)
```

Comment:
`--skip_baseline` is never checked in training. Also, it seems you've added support for skipping the baseline in `eval` but not mentioned it in the docstring.

Reply:
fixed